### PR TITLE
Support mounting volumes to containers

### PR DESCRIPTION
**Pull request by prachirp**
Fixes https://github.com/GoogleContainerTools/kpt/issues/359

### CODE REVIEW

**Comment by prachirp** [comment]
Fixes https://github.com/kubernetes-sigs/kustomize/issues/1902

File: kyaml/runfn/runfn.go [hunk 1]
```diff
@@ -51,6 +51,10 @@ type RunFns struct {
 	// NetworkName is the name of the docker network to use for the container
 	NetworkName string
 
+	// Volumes Volumes allows directories to be specified outside the configuration
```
Comment (pwittrock) [inline kyaml/runfn/runfn.go]: nit: Volumes only once

File: kyaml/kio/filters/functiontypes.go [hunk 2]
```diff
@@ -28,6 +28,9 @@ type FunctionSpec struct {
 
 	// Starlark is the spec for running a function as a starlark script
 	Starlark StarlarkSpec `json:"starlark,omitempty" yaml:"starlark,omitempty"`
+
+	// Volumes are the directories to mount as container volumes
+	Volumes []string `json:"volumes,omitempty" yaml:"volumes,omitempty"`
```
Comment (pwittrock) [inline kyaml/kio/filters/functiontypes.go]: We shouldn't need this

File: kyaml/runfn/runfn.go [hunk 3]
```diff
@@ -196,6 +207,24 @@ func (r RunFns) getFunctionsFromFunctionPaths() ([]kio.Filter, error) {
 	return r.getFunctionFilters(true, buff.Nodes...)
 }
 
+// getDirectoriesFromVolumes returns the set of directories read from r.Volumes
+// as a slice of Filters
+func (r RunFns) getDirectoriesFromVolumes() ([]kio.Filter, error) {
```
Comment (pwittrock) [inline kyaml/runfn/runfn.go]: Why do we need Volumes for this?

File: kyaml/runfn/runfn.go [hunk 4]
```diff
@@ -51,6 +51,10 @@ type RunFns struct {
 	// NetworkName is the name of the docker network to use for the container
 	NetworkName string
 
+	// Volumes Volumes allows directories to be specified outside the configuration
+	// directory.
+	Volumes []string
```
Comment (pwittrock) [inline kyaml/runfn/runfn.go]: Shouldn't we be using StorageMounts?

File: functions/examples/validator-kubeval/local-resource/example-use.yaml [hunk 5]
```diff
@@ -12,13 +12,6 @@ spec:
   strict: true
   ignoreMissingSchemas: true
 
-  # TODO: Remove these once function container network/volumes features are
-  # stabilized.
-  # Relevant issues:
-  #   - https://github.com/kubernetes-sigs/kustomize/issues/1901
-  #   - https://github.com/kubernetes-sigs/kustomize/issues/1902
-  kubernetesVersion: "1.16.0"
```
Comment (pwittrock) [inline functions/examples/validator-kubeval/local-resource/example-use.yaml:20]: I think we need to update the docs or something before removing these

File: kyaml/runfn/runfn.go (hunk 4 above)
Comment (prachirp) [reply kyaml/runfn/runfn.go]: Done

File: functions/examples/validator-kubeval/local-resource/example-use.yaml (hunk 5 above)
Comment (prachirp) [reply functions/examples/validator-kubeval/local-resource/example-use.yaml]: Updated TODO

File: kyaml/runfn/runfn.go (hunk 3 above)
Comment (prachirp) [reply kyaml/runfn/runfn.go]: Using storage mounts now

File: kyaml/kio/filters/container.go [hunk 6]
```diff
@@ -156,25 +156,31 @@ func (c ContainerFilter) String() string {
...
-
 func (s *StorageMount) String() string {
 	return fmt.Sprintf("type=%s,src=%s,dst=%s:ro", s.MountType, s.Src, s.DstPath)
 }
 
+func StringToStorageMount(s string) StorageMount {
+	m := make(map[string]string)
+	options := strings.Split(s, ",")
+	for _, option := range options {
+		keyVal := strings.Split(option, "=")
```
Comment (pwittrock) [inline kyaml/kio/filters/container.go:167]: nit: `SplitN(...., 2)`

Comment (prachirp) [reply kyaml/kio/filters/container.go]: Done

**Pull request merged**